castervoice/rules/core/keyboard_rules/keyboard.py

```diff
@@ -0,0 +1,147 @@
+from dragonfly import Function, Repeat, Dictation, Choice, ContextAction
+from castervoice.lib.context import AppContext
+
+from castervoice.lib import navigation, context, textformat, text_utils
+from castervoice.rules.core.navigation_rules import navigation_support
+from dragonfly.actions.action_mimic import Mimic
+
+from castervoice.lib.actions import Key, Mouse
+from castervoice.rules.ccr.standard import SymbolSpecs
+
+try:  # Try first loading from caster user directory
+    from alphabet_rules.alphabet_support import caster_alphabet
+except ImportError: 
+    from castervoice.rules.core.alphabet_rules.alphabet_support import caster_alphabet
+
+try:  # Try  first loading  from caster user directory
+    from punctuation_rules.punctuation_support import double_text_punc_dict, text_punc_dict
+except ImportError: 
+    from castervoice.rules.core.punctuation_rules.punctuation_support import double_text_punc_dict, text_punc_dict
+
+from castervoice.lib.const import CCRType
+from castervoice.lib.ctrl.mgr.rule_details import RuleDetails
+from castervoice.lib.merge.additions import IntegerRefST
+from castervoice.lib.merge.mergerule import MergeRule
+from castervoice.lib.merge.state.actions import AsynchronousAction, ContextSeeker
+from castervoice.lib.merge.state.actions2 import UntilCancelled
+from castervoice.lib.merge.state.short import S, L, R
+
+_tpd = text_punc_dict()
+
+class Keyboard(MergeRule):
+    pronunciation = "keyboard"
+    
+    mapping = {
+        "<modifier> <button_dictionary_1>":
+              R(Key("%(modifier)s%(button_dictionary_1)s"),
+              rdescript="press modifiers plus buttons from button_dictionary_1, non-repeatable"),
+        "<hold_release> [<modifier>] <button_dictionary_1>":
+              R(Key("%(modifier)s%(button_dictionary_1)s:%(hold_release)s"),
+              rdescript="press modifiers plus buttons from button_dictionary_1, non-repeatable"),
+    }
+
+    # These buttons can be used without using the "press" prefix.
+    right_spec = "(right | ross)"
+    left_spec = "(left | lease)"
+    button_dictionary_1 = {
+        "(F{}".format(i) + " | function {})".format(i) : "f{}".format(i)
+        for i in range(1, 13)
+    }
+    button_dictionary_1.update(caster_alphabet())
+    button_dictionary_1.update(_tpd)
+    longhand_punctuation_names = {
+        "minus": "hyphen",
+        "hyphen": "hyphen",
+        "comma": "comma",
```

Comment:
In order for natlink to recognize comma symbol the following is needed
```
if (dragonfly.get_current_engine().name == 'natlink'):
    comma = "(comma | ,)"
else:
    comma = "comma"
```
`comma: ",",`